### DISCUSSION TITLE
I wrote an example sketch to test the USB host with Arduino.

### DISCUSSION THREAD

**Discussion by HisashiKato** [sekigon-gonnoc/Pico-PIO-USB#75] [Show and tell]
Hello.
I wrote an example sketch to test a USB host using Pico-PIO-USB and Adafruit TinyUSB Library for Arduino.

This is a ZIP file of sketches.
[usb_host_test.zip](https://github.com/sekigon-gonnoc/Pico-PIO-USB/files/11155901/usb_host_test.zip)

https://user-images.githubusercontent.com/68432957/230003007-780da932-741f-4cac-bbfb-68de2b236d6e.mp4

This is a movie taken while moving.
In this movie, I'm using Adafruit QT Py RP2040, so I'm using GPIO 28(A1) for USB D+ and GPIO 29(A0) for USB D-.
 
You can demonstrate USB keyboard, USB mouse and PS4 DUALSHOCK4.

I'm an amateur so I'm not sure if this is correct.
Please verify it, rewrite it if it's wrong, and use it as an example if you like it.

Thank you.

(This sentence was translated from Japanese to English using a translation site)

**Comment by touchgadget** [comment]
I am doing something very similar using the Dualshock 4. Looks good to me but I am not an expert on this.
